src/raftstore/store/store.rs

```diff
@@ -1062,12 +1062,20 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                 // In this worst case scenario, the new split raft group will not be available
                 // since there is no leader established during one election timeout after the split.
                 let is_leader = self.region_peers[&region_id].is_leader();
-                if is_leader && right.get_peers().len() > 1 {
+                let my_id = new_peer.peer_id();
+                if right.get_peers().into_iter().all(|p| p.get_id() >= my_id) {
+                    if right.get_peers().len() > 1 {
+                        new_peer.raft_group.raft.term -= 1;
+                        // Use a smaller term so the assigned leader won't have lease.
+                        new_peer.mut_store().applied_index_term -= 1;
+                        new_peer.raft_group.raft.become_candidate();
+                        new_peer.raft_group.raft.become_leader();
```

Comment:
I think we must keep the same way with Raft paper, for a term if there is a Leader, other followers must all vote for it already.